src/socket.cr

```diff
@@ -363,6 +363,11 @@ class Socket < IO
 
   protected def recvfrom(bytes)
     sockaddr = Pointer(LibC::SockaddrStorage).malloc.as(LibC::Sockaddr*)
+    # initialize sockaddr with the initialized family of the socket
+    copy = sockaddr.value
+    copy.sa_family = family
+    sockaddr.value = copy
```

Comment:
Does `sockaddr.value = sockaddr.value.tap { |copy| copy.sa_family = family }` read nicer?

Reply:
Or even shorter:

```cr
sockaddr.value = sockaddr.value.tap(&.sa_family = family)
```

Reply:
I always forget you can chain like that with `&.`, unlike in Ruby 😄

Reply:
Yeah, you can write it like that. But it's not even shorter in terms of overall characters. It's just cramped in a single line. Is it more readable? In doubt I'd rather prefer the less-fancy multiline assignment. It should be easier to understand.

Reply:
@straight-shoota I personally think the inline version makes it clearer that we are replacing the existing value with the same value but modified slightly.  I didn't recognize that quickly on first glance with the original code, whereas my eyes understood that better in the latter. (I think because the `value = value + stuff` pattern is relatively common and my eyes see that pattern first...maybe it's just me 😆)

Reply:
It's a struct so you'll be modifying a copy.

Please, let's. ot waste time discussing such little details, especially when they can break the code. If it ain't broken don't fix it.